storage/types_test.go

```diff
@@ -147,6 +274,70 @@ func TestSplit(t *testing.T) {
 	}
 }
 
+func TestSplitPanics(t *testing.T) {
+	n, err := NewNodeIDForTreeCoords(8, 4567, 64)
+	if err != nil {
+		t.Fatal(err)
+	}
+	for _, tc := range []struct {
+		name      string
+		pBytes    int
+		sBits     int
+		wantPanic bool
+	}{
+		{
+			name:   "split8bits",
+			pBytes: 6,
+			sBits:  8,
+		},
+		{
+			name:   "split16bits",
+			pBytes: 5,
+			sBits:  16,
+		},
+		{
+			name:      "split15bits",
+			pBytes:    5,
+			sBits:     15,
+			wantPanic: true, // There's 16 bits left, which is more than requested.
+		},
+		{
+			name:      "splitnothing",
+			pBytes:    7, // This is 56 bits - same as the PrefixLength so nothing left.
+			sBits:     0,
+			wantPanic: true,
+		},
+		{
+			name:      "split1bit",
+			pBytes:    7, // This is 56 bits - same as the PrefixLength so nothing left.
+			sBits:     1, // When there's nothing left any number of bits fails.
+			wantPanic: true,
+		},
+		{
+			name:      "splitoutofbounds",
+			pBytes:    9, // There aren't 72 bits in the path.
+			sBits:     0,
+			wantPanic: true,
+		},
+	} {
+		t.Run(fmt.Sprintf(tc.name), func(t *testing.T) {
+			// Unfortunately we have to test for panics.
+			defer func() {
+				got := recover()
+				if (got != nil && !tc.wantPanic) || (got == nil && tc.wantPanic) {
+					t.Errorf("Incorrect panic behaviour got: %v, want: %v", got, tc.wantPanic)
+				}
+			}()
+			_, _ = n.Split(tc.pBytes, tc.sBits)
+		})
+	}
+	// The 2 cases where split should panic are 1.) There are more bits left
+	// after the prefix point than have been requested. 2.) Trying to split zero
+	// bits off the end of the NodeID.
+	for b := 0; b < 64; b++ {
```

Comment:
This comment is helpful but the loop isn't....

Reply:
Oops. Didn't clean up properly after restructure. Fixed.